pandas/tests/io/parser/dtypes.py

```diff
@@ -114,6 +114,17 @@ def test_categorical_dtype(self):
         actual = self.read_csv(StringIO(data), dtype='category')
         tm.assert_frame_equal(actual, expected)
 
+    @pytest.mark.slow
+    def test_categorical_dtype_high_cardinality_numeric(self):
+        # GH 18186
+        data = sorted([str(i) for i in range(10**6)])
+        expected = pd.DataFrame({'a': Categorical(data, ordered=True)})
+        actual = self.read_csv(StringIO('a\n' + '\n'.join(data)),
+                               dtype='category')
+        actual.a.cat.reorder_categories(sorted(actual.a.cat.categories),
+                                        ordered=True, inplace=True)
```

Comment:
Can you do this by assignment, i.e. `actual['a'] = ...`, instead of `inplace`.  The convention is generally to avoid using `inplace` in tests.

Reply:
I will make this change but I did the inplace hoping it would be faster and more memory efficient. Can you justify why it is preferred not to do inplace?

Reply:
@sam-cohan ``inplace=True`` is not more efficient in any way. Furthermore it is much harder to read, we do not use it in the codebase except in support of specific API where it is user supported.

Reply:
sorting should be using ``np.sort`` here

Reply:
Done on np.sort. 
I see your point about `inplace=True` in an ideal world, but FYI, in my workflows I typically use this to avoid temporarily needing double the storage for sorting (since I am often playing close to limits of available physical memory). Would have been nice if assignment was smart enough to be memory efficient.

Reply:
To expand on `inplace=True` a bit:  Per my understanding, for most operations `inplace=True` essentially does the same thing as assignment.  It will still operate on a copy, but with the top-level reference being reassigned, so there usually isn't actually a gain in terms of memory efficiency, etc.

Reply:
Thanks for the note... that is unfortunate. Perhaps that is worthy of discussion in another venue.